### PR TITLE
Fixes 1130: toast on failed file upload

### PR DESCRIPTION
**Pull request by jlsherrill**
To Test:

1.  Try to drag and drop a file that is bigger than ~8096 bytes
2.  Try to upload a file via the file picker that is bigger than ~8096 bytes
3.  Try to drag and drop more than one file
4.  Try to think of other issues?

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-1130

**Comment by jlsherrill** [comment]
TODO add a test

**Review by rverdile — COMMENTED** [review]
Uploading was working for me sometimes. But if I upload a file that I expect to be invalid, I seem to be getting this error, which I know you had been seeing earlier.

> Error: Invalid hook call. Hooks can only be called inside of the body of a function component.

Do you see this too? For example, uploading a file created with `head -c 7M < /dev/urandom > myfile`, gives me that error.

**Comment by jlsherrill** [comment]
ahh, a refactor seemed to break this... let me look :)

**Comment by jlsherrill** [comment]
should be fixed now

**Review by rverdile — COMMENTED** [review]
looks like it's working now. There's also some linter issues, you can see them locally with `npm run lint`

File: src/Pages/ContentListTable/components/AddContent/AddContent.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {
+    AlertVariant,
```
Comment (rverdile) [inline src/Pages/ContentListTable/components/AddContent/AddContent.tsx:2]: there's also a linter warnining for this being unused

File: src/Pages/ContentListTable/components/EditContentModal/EditContentModal.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import {
+    AlertVariant,
```
Comment (rverdile) [inline src/Pages/ContentListTable/components/EditContentModal/EditContentModal.tsx:2]: here as well

File: src/Pages/ContentListTable/components/AddContent/AddContent.tsx (hunk 1 above)
Comment (jlsherrill) [reply src/Pages/ContentListTable/components/AddContent/AddContent.tsx:2]: strange, the linter was passing, it caught others of these

Comment (jlsherrill) [reply src/Pages/ContentListTable/components/AddContent/AddContent.tsx]: but good catch!

File: src/Pages/ContentListTable/components/AddContent/helpers.ts [hunk 3]
```diff
@@ -100,3 +102,18 @@ export const makeValidationSchema = () => {
       .uniqueProperty('url', 'Url\'s must be unique'),
   );
 };
+
+export const maxUploadSize = 8096;
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/AddContent/helpers.ts:106]: We usually do constants like: CAPITALS_THINGS

File: src/Pages/ContentListTable/components/AddContent/helpers.test.ts [hunk 4]
```diff
@@ -70,3 +70,22 @@ it('mapValidationData', () => {
   expect(mapValidationData([], [])).toEqual([]);
   expect(mapValidationData(validationData, formikErrors)).toEqual(success);
 });
+
+it('Notifies on file upload failure due to size', () => {
+    const notif = jest.fn((payload) => payload);
+    const f = new File([''], 'filename', { type: 'text/html'});
```
Comment (Andrewgdewar) [inline src/Pages/ContentListTable/components/AddContent/helpers.test.ts:76]: Ideally use a meaningful name instaed of "f". 
Although after working with Go recently I understand why you did that. lol

**Review by Andrewgdewar — APPROVED** [review]
Added a few #nits but otherwise looks great.

**Comment by swadeley** [comment]
Hello

I tried to drag and drop file of 24.8kb

I saw Red exclamation mark and popup message:
 "There was an problem uploading the file.
The file is larger than 8096 bytes."

Using the file upload browser I tried to upload a file of 1.3kb and it worked, i.e. it was allowed.

**Comment by swadeley** [comment]
Hi

I tired to drag and drop two small  files:
"Danger alert:There was an problem uploading the file.
Only a single file upload is supported."

The file upload tool does not allow me to select two files.

All good, thank you.

**Pull request merged**